lib/ansible_tower_client/base_models/job_template.rb

```diff
@@ -25,5 +37,19 @@ def extra_vars_hash
     def override_raw_attributes
       { :credential => :credential_id, :inventory => :inventory_id, :project => :project_id }
     end
+
+    private
+
+    def check_ignorable_options(options)
+      ignored_options = IGNORABLE_OPTIONS.select do |option, checkmark|
+        !try(checkmark) && options.slice(option.to_sym, option.to_s).any?(&:present?)
+      end.keys
+
+      return if ignored_options.empty?
+
+      ignored_options.map! { |option| ":#{option}" }
+      message = "Option#{'s' if ignored_options.size > 1} #{ignored_options.to_sentence} will be ignored"
```

Comment:
"Option will be ignored" sounds like a warning not an error. An error message like "Option is provided but corresponding ask on launch flag has not been turn on" is more helpful.